gpjax/mean_functions.py

```diff
@@ -25,16 +25,26 @@ def __call__(self, x: f64["N D"]) -> f64["N Q"]:
         """
         raise NotImplementedError
 
-    @property
     @abc.abstractmethod
-    def params(self) -> dict:
+    def _initialise_params(self, key: jnp.DeviceArray) -> Dict:
         """Return the parameters of the mean function. This method is required for all subclasses.
 
         Returns:
             dict: The parameters of the mean function.
         """
         raise NotImplementedError
 
+    def init(self, key: jnp.ndarray) -> Dict:
+        """Initialise the stateful parameters of the mean function. This method is required for all subclasses.
+
+        Args:
+            key (Array): The key used to initialise the stateful parameters.
+
+        Returns:
+            dict: The initialised stateful parameters of the mean function.
+        """
+        return self.params
```

Comment:
Like with the likelihoods, I'm not sure what this does given `.params` does not exist.